src/devices/Pcx857x/tests/ConstructionTests.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace Iot.Device.Pcx857x.Tests
+{
+    public class ConstructionTests : Pcx857xTest
+    {
+        [Theory]
+        [MemberData(nameof(TestDevices))]
+        public void RegisterInitialization(TestDevice testDevice)
+        {
+            // All pins should be set to output.
+            Pcx857x device = testDevice.Device;
+            if (device.PinCount == 8)
+            {
+                Assert.Equal(0x00, device.ReadByte());
+            }
+            else
```

Comment:
I'd add Assert `device.PinCount == 8 || device.PinCount == 16` or do explicit `else if` and `else` which throws

Reply:
Sure, added the assert.